File: Easy/divi9/divi9.java
//run as java -ea divi9
import java.util.Scanner;

public class divi9{
	public static void main(String[] args){
		int T,sum,M;
		String N;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=100000) : "T must be 1 <= T <= 10^5";

		while(T>0){
			N = scan.next();
			sum = 0;

			for(int i=0;i<N.length();i++){
				sum += (N.charAt(i)-'0');
			}

			M = sum%9;

			if((N.length()>1 && sum<9) || (M>(9-M))){
				System.out.println(9-M);
			}else{
				System.out.println(M);
			}

			T--;
		}
	}
}

File: Easy/divi9/divi9.c
#include <stdio.h>
#include <assert.h>
#include <string.h>

int main(void){
	int T,sum,M;
	char N[100005];

	scanf("%d",&T);
	assert(T>=1 && T<=100000);

	while(T>0){
		scanf("%s",N);
		sum = 0;

		for(int i=0;i<strlen(N);i++){
			sum += (N[i]-'0');
		}

		M = sum%9;

		if((strlen(N)>1 && sum<9) || (M>(9-M))){
			printf("%d\n",9-M);
		}else{
			printf("%d\n",M);
		}

		T--;
	}
	
	return 0;
}